### DISCUSSION TITLE
[Warning] AutoInterface[Default Interface] could not autoconfigure. This interface currently provides no connectivity.

### DISCUSSION THREAD

**Discussion by aj5r1gg** [markqvist/Reticulum#390] [Help & Questions] [answered]
New install of Nomadnet and I'm getting this warning:

[Warning] AutoInterface[Default Interface] could not autoconfigure. This interface currently provides no connectivity.

I have upgraded Reticulum to version 0.6.7, lxmf to version 0.3.8 and Nomadnet to version 0.4.2

How can I change this?

**Comment by faragher** [comment]
Are you on Windows? Was this an error that occurred after an upgrade?

I don't understand the AutoInterface very well, but I'm running 0.6.7 on multiple boxes without issue, and Windows doesn't currently support AutoInterface. If you're looking to link Windows boxes over your network, you can still use TCPServer and TCPClient with your IP addresses.

**Reply by aj5r1gg** [reply]
Hey @faragher, thanks for the response. No, I'm not running Windows. Chucked that out after it was causing my computer to randomly shutdown. Went to Ubuntu and haven't had an issue since.

Nonetheless, it was an error that was there before upgrading too.

**Reply by faragher** [reply]
I hear that. I have a lone Windows system for gaming and development, and I spent so much time gutting it so it'd stop doing stuff I didn't want. And I have a professional license that supposedly allows me to disable all that nonsense. Well now who's laughing?

Ahem. Sorry.

Okay, if the error pre-exists the upgrade, we can take that off the table. IIRC, the AutoInterface is an IPv6 based discovery mechanism. The nodes don't need to have addresses, but you do need IPv6 support on the router, or so I've been told. Have you seriously locked down a router with OpenWRT or the like? I shut down most of my IPv6 on this net, but it still works for me. Although now that I think about it, they may all be on a single subnet using a stock router.

**Reply by markqvist** [reply, marked as answer]
There doesn't need to be IPv6 support on the router, in fact there doesn't need to be a router at all. The minimum requirements for AutoInterface to work is:

- A wired or wireless ethernet-based network, with peers in the same local broadcast domain
- IPv6 support in the operating system of the peers, and IPv6 support enabled on the ethernet/wifi interface

In Ubuntu, the IPv6 criteria should be enabled by default, so unless you have manually turned it off, it's a bit weird that it is not working "out of the box".

So for example, something as simple as a an ethernet switch (even with no router, no DHCP, and no IP infrastructure) will work with AutoInterface.

Also, maybe an obvious thing, but just for the sake of making it clear, if you are using AutoInterface over a WiFi network, you need to have that WiFi interface active, by being connected to something (an access point or ad-hoc network). Reticulum does not currently control the WiFi hardware at the raw frame level (this requires root privileges, and would be more of a pain for users to set up), so you must be actually connected to "something", before Reticulum's AutoInterface can use the interface.

**Reply by aj5r1gg** [reply]
Yup, that was it. I had disabled ipv6. Thank you for your assistance.

**Reply by markqvist** [reply]
Good to hear!

For context, the reason we need to make use of the IPv6 functionality of the OS, is that direct access to Ethernet / WiFi devices requires elevated privileges or permissions in most operating systems. While just accessing the raw Ethernet/WiFi devices and formatting our own packets to carry Reticulum would be a more direct and less contrived way of getting AutoInterface functionality to work, it would be quite a nuisance for the user to have to run Reticulum as a root user or system service.

As such, the best middle way is to use the IPv6 functionality, which allows us essentially the same functionality, also when no IP infrastructure, routers or DHCP servers are present, but doesn't require any special privileges.
